app/assets/javascripts/discourse/app/templates/application.hbs

```diff
@@ -77,8 +77,13 @@
         />
       </div>
 
+      {{!-- {{#if this.showSidebarFromService}} --}}
+      {{! SIDEBAR ON (ERROR) }}
+      {{!-- {{/if}} --}}
```

Comment:
@davidtaylorhq can you give me a hand with this? I tried to emulate the same thing as `hide-application-footer` helper, but when it is used on a component rendered before the main `{{outlet}}`, it fails. 

If you pull this branch, it will work (you should see `SIDEBAR ON` output except under `/wizard`). But if you uncomment these lines here before the `outlet`, there will be an error: 

```
Uncaught (in promise) Error: Assertion Failed: You attempted to update `_value` on `TrackedStorageImpl`, but it had already been used previously in the same computation.  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, and performance issues, and is not supported.
```

<img width="1486" alt="image" src="https://github.com/discourse/discourse/assets/368961/a00e8dc4-13e3-4cb6-8de4-8716ced3b4f2">

Reply:
Hmm, this is a tricky one. I guess by the time your helper is called, `this.sidebarState.showSidebar` has already been checked. We tend not to have this problem in the footer, because it is rendered after everything else.

It pains me to say it but... maybe add a `next()` in the helper? So it would be

```
next(() => this.sidebarState.registerHider(this));
```
that way, the change will happen in the next runloop and Ember will un-render the sidebar.

(cc @cvx in case you have any better ideas?)